### PR TITLE
`ultralytics 8.1.24` new OpenVINO 2023.3 export updates

### PR DESCRIPTION
**Pull request by adrianboguszewski**
Changed the way YOLO is exported to OpenVINO (no more ONNX needed as an intermediate step)

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced support for OpenVINO export and import with version bump and PyTorch compatibility updates.

### 📊 Key Changes
- Updated the OpenVINO package requirement from `openvino-dev>=2023.0` to `openvino>=2023.3`.
- Added PyTorch version check, requiring PyTorch `1.13.0 or greater` for OpenVINO export.
- Modified OpenVINO export and import code to adapt to the new package requirements and functionalities.
- Incremented the Ultralytics version from `8.1.23` to `8.1.24`.

### 🎯 Purpose & Impact
- 🆕 Enables users to leverage the latest OpenVINO features and improvements for model optimization and deployment.
- 🧠 Ensures compatibility with newer PyTorch versions, improving stability and support.
- 🔍 Provides more reliable OpenVINO model export/import, potentially boosting performance for AI models on various platforms.
- 👥 Impacts users who utilize OpenVINO for exporting their models, requiring them to update their environment to continue using these features without issues.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@adrianboguszewski super cool, I like it! This might help exporting some newer architectures like YOLO-World that has modules that are currently not exportable to ONNX. 

It seems to not be supported in earlier versions of Python or Torch though, we'll need to run some tests to see what the cuttoff is and return a warning if OpenVINO export is attempted below this cutoff.

@Laughing-q

**Comment by adamp87** [comment]
@adrianboguszewski cool nice update!

I checked and the Conv of the DFL is not being ignored. The module is named as `__module.model.22.dfl.conv/aten::_convolution/Convolution`. Therefore I have changed [line 467](https://github.com/adrianboguszewski/ultralytics/blob/fd8fb926dccda2feb8d5ae1548bc33d6354152b8/ultralytics/engine/exporter.py#L467) to `f".*{head_module_name}\\.dfl.*",`. After that I am getting exactly the same results for COCO128.

I ran the tests with:
```
yolo export model=yolov8n.pt imgsz=640 format=openvino int8
yolo val model=yolov8n_int8_openvino_model data=coco128.yaml imgsz=640 int8
```

**Comment by adrianboguszewski** [comment]
@adamp87 Fixed. Thanks for the commands :)

**Comment by glenn-jocher** [comment]
@adrianboguszewski ok I've debugged our torch 1.8 test a bit since it was failing on this PR. It looks like the earliest compatible torch version is torch==1.13. This PR is the only change in the repo that would require 1.13, as our current lower compat limit is 1.8, so I think I'll leave the 1.8 in place and make a special check for openvino export

File: ultralytics/engine/exporter.py [hunk 1]
```diff
@@ -458,27 +456,32 @@ def transform_fn(data_item):
...
-                        f"/{head_module_name}/Add",
-                        f"/{head_module_name}/Sub",
-                        f"/{head_module_name}/Mul",
-                        f"/{head_module_name}/Div",
-                        f"/{head_module_name}/dfl",
+                        f".*{head_module_name}/.*/Add",
+                        f".*{head_module_name}/.*/Sub*",
+                        f".*{head_module_name}/.*/Mul*",
+                        f".*{head_module_name}/.*/Div*",
+                        f".*{head_module_name}\\.dfl.*",
```
Comment (glenn-jocher) [inline ultralytics/engine/exporter.py:471]: @adamp87 @adrianboguszewski guys I just want to check here that this handling of the DFL modules is correct. It's the only line that differs from the rest. Can you please verify this is ok?

Comment (adrianboguszewski) [reply ultralytics/engine/exporter.py:480]: This should be correct. The name differs a little bit e.g. `__module.model.22.dfl.conv/aten::_convolution/Convolution`

**Comment by adrianboguszewski** [comment]
> @adrianboguszewski ok I've debugged our torch 1.8 test a bit since it was failing on this PR. It looks like the earliest compatible torch version is torch==1.13. This PR is the only change in the repo that would require 1.13, as our current lower compat limit is 1.8, so I think I'll leave the 1.8 in place and make a special check for openvino export

That's ok, then.

File: ultralytics/nn/autobackend.py [hunk 2]
```diff
@@ -180,17 +180,17 @@ def __init__(
             metadata = session.get_modelmeta().custom_metadata_map  # metadata
         elif xml:  # OpenVINO
             LOGGER.info(f"Loading {w} for OpenVINO inference...")
-            check_requirements("openvino>=2023.0")  # requires openvino-dev: https://pypi.org/project/openvino-dev/
-            from openvino.runtime import Core, Layout, get_batch  # noqa
+            check_requirements("openvino>=2023.3")  # requires openvino-dev: https://pypi.org/project/openvino-dev/
```
Comment (adrianboguszewski) [inline ultralytics/nn/autobackend.py:183]:
```suggestion
            check_requirements("openvino>=2023.3")  # requires openvino: https://pypi.org/project/openvino-dev/
```

File: pyproject.toml [hunk 3]
```diff
@@ -98,7 +98,7 @@ dev = [
 export = [
     "onnx>=1.12.0", # ONNX export
     "coremltools>=7.0; platform_system != 'Windows' and python_version <= '3.11'", # CoreML supported on macOS and Linux
-    "openvino-dev>=2023.0; python_version <= '3.11'", # OpenVINO export
+    "openvino-dev>=2023.3; python_version <= '3.11'", # OpenVINO export
```
Comment (adrianboguszewski) [inline pyproject.toml:101]:
```suggestion
    "openvino>=2023.3; python_version <= '3.11'", # OpenVINO export
```

**Comment by glenn-jocher** [comment]
@adrianboguszewski can we remove `openvino-dev` completely from the requirements in this PR as in the related OpenVINO PR just opened in https://github.com/ultralytics/ultralytics/pull/8638?

File: ultralytics/engine/exporter.py [hunk 4]
```diff
@@ -411,16 +411,16 @@ def export_onnx(self, prefix=colorstr("ONNX:")):
     @try_export
     def export_openvino(self, prefix=colorstr("OpenVINO:")):
         """YOLOv8 OpenVINO export."""
-        check_requirements("openvino-dev>=2023.0")  # requires openvino-dev: https://pypi.org/project/openvino-dev/
-        import openvino.runtime as ov  # noqa
-        from openvino.tools import mo  # noqa
+        check_requirements("openvino-dev>=2023.3")  # requires openvino-dev: https://pypi.org/project/openvino-dev/
```
Comment (adrianboguszewski) [inline ultralytics/engine/exporter.py:414]:
```suggestion
        check_requirements("openvino>=2023.3")  # requires openvino: https://pypi.org/project/openvino-dev/
```

**Comment by adrianboguszewski** [comment]
Yes, I did that. We're ready to merge.

**Comment by glenn-jocher** [comment]
Hi there! It sounds like you're encountering an issue with OpenVINO on Node.js. Could you please provide more details or error logs? This will help us understand the problem better and find a solution. Also, ensure that your OpenVINO and Node.js environments are set up correctly and are compatible with each other. Thanks for bringing this to our attention! 🛠️

**Pull request merged**